Holovibes/cuda_includes/otsu.cuh

```diff
@@ -1,45 +1,52 @@
 /*! \file
  *
- * \brief #TODO Add a description for this file
+ * \brief function for otsu comput
  */
 #pragma once
 
 #include "frame_desc.hh"
 
+using uint = unsigned int;
+
 /**
  * \brief Compute Binarisation with Otsu threshold
  *
  * \param[in out] input The image to process
+ * \param[out] histo_buffer_d gpu buffer for histogram
  * \param[in] width Width of the frame
  * \param[in] height Height of the frame
  * \param[in] stream The CUDA stream on which to launch the operation
  */
-void compute_binarise_otsu(float* input, const size_t width, const size_t height, const cudaStream_t stream);
+void compute_binarise_otsu(
+    float* input, uint* histo_buffer_d, const size_t width, const size_t height, const cudaStream_t stream);
 
 /*! \brief Compute Binarisation with Otsu threshold and bradley method
  *
  * \param[in] d_image Input data should be contiguous
  * \param[out] d_output Where to store the output
+ * \param[out] histo_buffer_d gpu buffer for histogram
  * \param[in] width Width of the frame
  * \param[in] height Height of the frame
- * \param[in] window_size //TODO
- * \param[in] local_threshold_factor //TODO
+ * \param[in] window_size size of the windows
+ * \param[in] local_threshold_factor local threshold factor
  * \param[in] stream The CUDA stream on which to launch the operation
  */
 void compute_binarise_otsu_bradley(float* d_image,
                                    float*& d_output,
+                                   uint* histo_buffer_d,
                                    const size_t width,
                                    const size_t height,
                                    const int window_size,
                                    const float local_threshold_factor,
                                    const cudaStream_t stream);
 
-/*! \brief Normalistation of \d_image
+/*!
+ * \brief get otsu threshold
  *
- * \param[in out] d_image Data should be contiguous
- * \param[in] min Minimal value of \d_image
- * \param[in] max Maximal value of \d_image
+ * \param[in] d_image Input data should be contiguous
+ * \param[out] histo_buffer_d gpu buffer for histogram
  * \param[in] size Size of the frame
  * \param[in] stream The CUDA stream on which to launch the operation
+ * \return ostus threshold
  */
```

Comment:
Maybe explain what is the otsu threshold. The bufferd_image sould be named input_output.